src/hotspot/share/opto/mulnode.cpp

```diff
@@ -1506,8 +1506,9 @@ const Type* RotateLeftNode::Value(PhaseGVN* phase) const {
       return r1;
     }
     if (r1->is_con() && r2->is_con()) {
-      int shift = r2->get_con() & (BitsPerJavaInteger - 1); // semantics of Java shifts
-      return TypeInt::make((r1->get_con() << shift) | (r1->get_con() >> (32 - shift)));
+      unsigned int r1_con = (unsigned int)r1->get_con();
+      unsigned int shift = (unsigned int)(r2->get_con()) & (unsigned int)(BitsPerJavaInteger - 1); // semantics of Java shifts
```

Comment:
I'd prefer to use juint which actually is the same, but would be more consistent IMHO.

Reply:
+1

Reply:
I agree, I'll change it to `juint`